iothub/device/src/IotHubModuleClient.cs

```diff
@@ -170,10 +170,10 @@ public async Task SendTelemetryAsync(string outputName, TelemetryMessage message
         /// <returns>The task containing the event</returns>
         /// <exception cref="InvalidOperationException">Thrown if IotHubModuleClient instance is not opened already.</exception>
         /// <exception cref="OperationCanceledException">Thrown when the operation has been canceled.</exception>
-        public async Task SendTelemetryBatchAsync(string outputName, IEnumerable<TelemetryMessage> messages, CancellationToken cancellationToken = default)
+        public async Task SendMessagesToRouteAsync(string outputName, IEnumerable<TelemetryMessage> messages, CancellationToken cancellationToken = default)
```

Comment:
Are we deliberately keeping the name the same for single message and batch message?

Reply:
Ah, I missed the "s" at the end of "message"

Reply:
The API to send TelemetryMessage to IoT Hub directly is called "SendTelemetryAsync". Should we use the term "Telemetry" instead of "Message" in here as well? The data type involved is a "TelemetryMessage" in both cases.

Reply:
@rido-min doesn't see these messages as telemetry

Reply:
My understanding of Telemetry (in the context of IoTHub), is for MQTT messages published (D2C) to `devices/{clientId}/messages/events/` that can be routed to a configured BuiltIn-Endpoint/Route by using a Data source of type `Device Telemetry Messages`

My suggestion of changing the name of this overload is based on the assumption that D2C messages sent with this overload won't be able to be routed as Telemetry.. Let me run a quick test and I'll get back to this thread.

Reply:
just checked the behavior in hub with a device (non-edge) module.

The value used in `outputName` is part of the message metadata as:

```json
 "properties": {
            "application": {
                "$.on": "myOutput"
            }
        },
```

let me review the full edge module scenario, including routing.. maybe SendTelemetry is not so bad..

Reply:
also wondering if DeviceClient should also accept an output name

Reply:
What is the user scenario for adding `outputName` parameter to IotHubDeviceClient telemetry API?

Reply:
I dont know the use case, just trying to reflect the IoTHub features wrt messaging to have consistency in Device/Module clients.

Reply:
I've added the property to `TelemetryMessage` to allow someone to set it when sending from a device client, but I wouldn't go so far as to add a parameter for it as it is not a mainline scenario. Does that work for you?